main.py

```
@@ -31,16 +30,12 @@ def get_weather():
         if response.status_code == 200:
             weather_json = ujson.loads(response.text)
             current_time = time.localtime()
-            year, month, day, hour, minute, second, weekday, yearday = current_time
-            formatted_time = "{:04d}-{:02d}-{:02d}T{:02d}:00".format(year, month, day, hour)
-            weather_code = 0
-            wc_index = 0
-            for hours in weather_json["hourly"]["time"]:
+            year, month, day, hour, _, _, _, _ = current_time
+            formatted_time = f"{year:04d}-{month:02d}-{day:02d}T{hour:02d}:00"
+            for idx, hours in enumerate(weather_json["hourly"]["time"]):
                 if hours == formatted_time:
-                    break
-                wc_index += 1
-            weather_code = weather_json["hourly"]["weather_code"][wc_index]
-            return WEATHER_CODES[str(weather_code)]
+                    return WEATHER_CODES[str(weather_json["hourly"]["weather_code"][idx])]
+            return WEATHER_CODES["0"]   
```

Maybe better to return an error than an invalid element or undefined element

---

done

---

done